pkg/component/autoscaling/vpa/crd.go

```diff
@@ -71,3 +79,46 @@ func (v *vpaCRD) Destroy(ctx context.Context) error {
 
 	return nil
 }
+
+var (
+	// IntervalWaitForCRD is the interval used while waiting for the CRDs to become healthy
+	// or deleted.
+	IntervalWaitForCRD = 1 * time.Second
+	// TimeoutWaitForCRD is the timeout used while waiting for the CRDs to become healthy
+	// or deleted.
+	TimeoutWaitForCRD = 15 * time.Second
+	// Until is an alias for retry.Until. Exposed for tests.
+	Until = retry.Until
+)
+
+// Wait signals whether a CRD is ready or needs more time to be deployed.
+func (v *vpaCRD) Wait(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForCRD)
+	defer cancel()
+
+	return retry.Until(timeoutCtx, IntervalWaitForCRD, func(ctx context.Context) (done bool, err error) {
+		for _, resource := range crdResources {
+			r := resource
+			crd := &v1.CustomResourceDefinition{}
+
+			obj, err := kubernetes.NewManifestReader([]byte(r)).Read()
+			if err != nil {
+				return retry.SevereError(err)
+			}
+
+			if err := v.client.Get(ctx, client.ObjectKeyFromObject(obj), crd); client.IgnoreNotFound(err) != nil {
+				return retry.SevereError(err)
+			}
+
+			if err := health.CheckCustomResourceDefinition(crd); err != nil {
+				return retry.MinorError(err)
+			}
+		}
+		return retry.Ok()
+	})
```

Comment:
Can you put this `func` somewhere to the utils to make it reusable? Now, it's copied and duplicated too often.